lib/api/model/widget.dart

```diff
@@ -0,0 +1,131 @@
+import '../../log.dart';
+import '../../model/store.dart';
+import 'submessage.dart';
+
+/// States of a poll Zulip widget.
+///
+/// See also:
+/// - https://zulip.com/help/create-a-poll
+/// - https://github.com/zulip/zulip/blob/304d948416465c1a085122af5d752f03d6797003/web/shared/src/poll_data.ts
+class Poll {
```

Comment:
Let's avoid having a file named `widget.dart` that isn't about Flutter widgets. The term "widget" is a highly salient one in our codebase, with a particular meaning; so it's confusing to have the term appear with a different meaning.

Where we have to use it (to refer to the Zulip API concept), we can manage that by making longer names that clearly namespace the reference into the context of the Zulip poll/todo "widget" concept. For this filename, though… how about just putting this all inside `submessage.dart`? It seems like most of that file is specific to polls anyway.